Add npm prepare script to allow install via git URI

thinkh opened the pull request:
Related to https://github.com/datavisyn/lineup_app/issues/15

**prerequisites**: 
 * [x] branch is up-to-date with the branch to be merged with, i.e. develop
 * [x] build is successful
 * [x] code is cleaned up and formatted 


### Summary
 
Adding the `prepare` script allows to install this repo from git URI:

```
npm install --save git+https://github.com/datavisyn/lineupjs.git
```

Or for instance by modifying the dependency in the `package.json` of [lineup_app](https://github.com/datavisyn/lineup_app/): 

```
  "dependencies": {
    "lineupjs": "github:Caleydo/lineupjs#develop",
  }
```

For further information see:

* https://docs.npmjs.com/misc/scripts
* https://github.com/npm/npm/issues/3055
* Example: https://github.com/tasti/react-linkify/pull/61


**NOTE!**
Using the git URI as dependency will build the LineUp library on `npm install`. Then the whole install process will take longer than usual. It might appear that the installation process is stuck, but it is not! So be patient or run `npm install -logLevel verbose` to get further information about the progress.

sgratzl commented:
* why are there so many commit for this single change?
 * what happens upon installing the published library?
 * why does `npm run build` work? I mean the dev dependencies aren't installed, are they?
 * what happens upon publishing the library

sgratzl commented:
> According to the npm documentation it will also run the build script:

that won't work, since the published version doesn't contain the source build anymore. and it shouldn't create a new build each time when someone installs the library

sgratzl commented:
> I don't know. It seems to happen, after I merged the develop branch from datavisyn into Caleydo.

please rebranch from the original repo and just change the one commit

thinkh commented:
> that won't work, since the published version doesn't contain the source build anymore. and it shouldn't create a new build each time when someone installs the library

I would like to know if this really happens. How can I test this?

thinkh commented:
> please rebranch from the original repo and just change the one commit

Ok, let us finish the discussion here first. When we have a result I will do so.

thinkh commented:
@sgratzl Do you think the `prepack` hook is the better option according to the [npm documentation](https://docs.npmjs.com/misc/scripts) and https://stackoverflow.com/a/49292495/940219 ?

> prepack: run BEFORE a tarball is packed (on npm pack, npm publish, and when installing git dependencies)

thinkh commented:
So published a [test package](https://www.npmjs.com/package/thinkh-lineupjs-test) with the `prepare` script and installed it locally.

```
PS D:\dev\test> npm install thinkh-lineupjs-test
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN test@1.0.0 No description
npm WARN test@1.0.0 No repository field.

+ thinkh-lineupjs-test@3.0.1-beta.13
added 5 packages from 12 contributors and audited 5 packages in 1.428s
found 0 vulnerabilities
```

No build was started during the installation process and only the dependencies are installed: 

```
    Verzeichnis: D:\dev\test\node_modules


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       13.06.2018     17:23                @types
d-----       13.06.2018     17:23                lineupengine
d-----       13.06.2018     17:23                popper.js
d-----       13.06.2018     17:23                reflect-metadata
d-----       13.06.2018     17:23                thinkh-lineupjs-test
```

Then I changed the dependency in the package.json to:

```
  "dependencies": {
    "thinkh-lineupjs-test": "github:thinkh/lineupjs#thinkh/npm-dependency-test"
  }
```

Running `npm install` again will start the build process.

**Conclusion**

In my opinion the `prepare` script works well.
@sgratzl If you agree with the solution, I will file you a new PR with only this commit.

thinkh commented:
```
PS D:\dev\thinkh-lineupjs> npm run release:pre

> thinkh-lineupjs-test@3.0.1-beta.13 release:pre D:\dev\thinkh-lineupjs
> npm version prerelease && npm publish --tag=next && git push --follow-tags


> thinkh-lineupjs-test@3.0.1-beta.14 preversion D:\dev\thinkh-lineupjs
> npm test


> thinkh-lineupjs-test@3.0.1-beta.13 pretest D:\dev\thinkh-lineupjs
> npm run clean && npm run compile


> thinkh-lineupjs-test@3.0.1-beta.13 clean D:\dev\thinkh-lineupjs
> rimraf build dist && npm run clean:compile


> thinkh-lineupjs-test@3.0.1-beta.13 clean:compile D:\dev\thinkh-lineupjs
> rimraf src/**/*.map src/**/*.js src/**/*.d.ts tests/**/*.js tests/**/*.map tests/**/*.d.ts demo/**/*.js demo/**/*.map demo/**/*.d.ts


> thinkh-lineupjs-test@3.0.1-beta.13 compile D:\dev\thinkh-lineupjs
> tsc


> thinkh-lineupjs-test@3.0.1-beta.13 test D:\dev\thinkh-lineupjs
> node -e "process.exit(require('fs').existsSync('./tests')?1:0)" || karma start


> thinkh-lineupjs-test@3.0.1-beta.13 posttest D:\dev\thinkh-lineupjs
> npm run lint


> thinkh-lineupjs-test@3.0.1-beta.13 lint D:\dev\thinkh-lineupjs
> tslint --project tsconfig.json -c tslint.json && stylelint src/**/*.scss

v3.0.1-beta.14

> thinkh-lineupjs-test@3.0.1-beta.14 prepare .
> npm run build


> thinkh-lineupjs-test@3.0.1-beta.14 prebuild D:\dev\thinkh-lineupjs
> npm run clean && (node -e "process.exit(process.env.PHOVEA_SKIP_TESTS === undefined?1:0)" || npm run test)


> thinkh-lineupjs-test@3.0.1-beta.14 clean D:\dev\thinkh-lineupjs
> rimraf build dist && npm run clean:compile


> thinkh-lineupjs-test@3.0.1-beta.14 clean:compile D:\dev\thinkh-lineupjs
> rimraf src/**/*.map src/**/*.js src/**/*.d.ts tests/**/*.js tests/**/*.map tests/**/*.d.ts demo/**/*.js demo/**/*.map demo/**/*.d.ts


> thinkh-lineupjs-test@3.0.1-beta.14 pretest D:\dev\thinkh-lineupjs
> npm run clean && npm run compile


> thinkh-lineupjs-test@3.0.1-beta.14 clean D:\dev\thinkh-lineupjs
> rimraf build dist && npm run clean:compile


> thinkh-lineupjs-test@3.0.1-beta.14 clean:compile D:\dev\thinkh-lineupjs
> rimraf src/**/*.map src/**/*.js src/**/*.d.ts tests/**/*.js tests/**/*.map tests/**/*.d.ts demo/**/*.js demo/**/*.map demo/**/*.d.ts


> thinkh-lineupjs-test@3.0.1-beta.14 compile D:\dev\thinkh-lineupjs
> tsc


> thinkh-lineupjs-test@3.0.1-beta.14 test D:\dev\thinkh-lineupjs
> node -e "process.exit(require('fs').existsSync('./tests')?1:0)" || karma start


> thinkh-lineupjs-test@3.0.1-beta.14 posttest D:\dev\thinkh-lineupjs
> npm run lint


> thinkh-lineupjs-test@3.0.1-beta.14 lint D:\dev\thinkh-lineupjs
> tslint --project tsconfig.json -c tslint.json && stylelint src/**/*.scss


> thinkh-lineupjs-test@3.0.1-beta.14 build D:\dev\thinkh-lineupjs
> webpack --mode production

Starting type checking service...
Using 1 worker with 2048MB memory limit
Hash: bf3b3cc3c75033fdef60
Version: webpack 4.8.3
Time: 14930ms
Built at: 2018-06-13 17:38:04
                               Asset      Size  Chunks                    Chunk Names
248608643408d014c5148346db87791b.eot  20.5 KiB          [emitted]
b3646da949d9b249a801d1c515293f03.svg  22.6 KiB          [emitted]
b2e751c0a3ee1cfd9b19f52e9747488f.ttf  20.3 KiB          [emitted]
                         LineUpJS.js   478 KiB       0  [emitted]  [big]  LineUpJS
                        LineUpJS.css  89.5 KiB       0  [emitted]         LineUpJS
Entrypoint LineUpJS [big] = LineUpJS.js LineUpJS.css
 [4] ./src/internal/math.ts 6.81 KiB {0} [built]
[13] ./src/provider/ADataProvider.ts 19.9 KiB {0} [built]
[34] ./src/interfaces.ts 111 bytes {0} [built]
[37] ./src/ui/LineUp.ts 2.42 KiB {0} [built]
[41] ./src/provider/LocalDataProvider.ts 7.91 KiB {0} [built]
[44] ./src/renderer/index.ts + 31 modules 102 KiB {0} [built]
     | ./src/renderer/BooleanCellRenderer.ts 930 bytes [built]
     | ./src/renderer/index.ts 4.51 KiB [built]
     | ./src/renderer/StringCellRenderer.ts 5.2 KiB [built]
     | ./src/renderer/AnnotationRenderer.ts 1.39 KiB [built]
     | ./src/renderer/VerticalBarCellRenderer.ts 3.29 KiB [built]
     | ./src/renderer/BarCellRenderer.ts 2.55 KiB [built]
     | ./src/renderer/DefaultCellRenderer.ts 982 bytes [built]
     | ./src/renderer/ActionRenderer.ts 2 KiB [built]
     | ./src/renderer/BoxplotCellRenderer.ts 8.11 KiB [built]
     | ./src/renderer/BrightnessCellRenderer.ts 2.16 KiB [built]
     | ./src/renderer/CategoricalCellRenderer.ts 6.79 KiB [built]
     | ./src/renderer/UpSetCellRenderer.ts 4.68 KiB [built]
     | ./src/renderer/CategoricalHeatmapCellRenderer.ts 3.59 KiB [built]
     | ./src/renderer/CategoricalStackedDistributionlCellRenderer.ts 3.92 KiB [built]
     | ./src/renderer/CircleCellRenderer.ts 1.65 KiB [built]
     |     + 17 hidden modules
[54] ./src/model/ReduceColumn.ts 4.07 KiB {0} [built]
[68] ./src/ui/taggle/Taggle.ts 3.61 KiB {0} [built]
[75] ./src/config.ts 865 bytes {0} [built]
[79] ./src/model/index.ts + 10 modules 33.3 KiB {0} [built]
     | ./src/model/CategoricalMapColumn.ts 1.93 KiB [built]
     | ./src/model/index.ts 7.57 KiB [built]
     | ./src/model/CategoricalsColumn.ts 1.59 KiB [built]
     | ./src/model/DateColumn.ts 1.71 KiB [built]
     | ./src/model/DatesColumn.ts 3.33 KiB [built]
     | ./src/model/BooleansColumn.ts 1.4 KiB [built]
     | ./src/model/NumberMapColumn.ts 5.71 KiB [built]
     | ./src/model/SetColumn.ts 5.43 KiB [built]
     | ./src/model/StringsColumn.ts 2.25 KiB [built]
     | ./src/model/DummyColumn.ts 555 bytes [built]
     | ./src/model/DatesMapColumn.ts 1.59 KiB [built]
[86] ./src/provider/index.ts + 1 modules 3.68 KiB {0} [built]
     | ./src/provider/index.ts 337 bytes [built]
     | ./src/provider/RemoteDataProvider.ts 3.32 KiB [built]
[87] ./src/builder/index.ts + 16 modules 45 KiB {0} [built]
     | ./src/builder/column/DateColumnBuilder.ts 675 bytes [built]
     | ./src/builder/index.ts 794 bytes [built]
     | ./src/builder/DataBuilder.ts 3.84 KiB [built]
     | ./src/builder/column/index.ts 281 bytes [built]
     | ./src/builder/adapter/index.ts 1.81 KiB [built]
     | ./src/builder/column/NumberColumnBuilder.ts 3.43 KiB [built]
     | ./src/builder/column/StringColumnBuilder.ts 961 bytes [built]
     | ./src/builder/RankingBuilder.ts 10.3 KiB [built]
     | ./src/builder/LineUpBuilder.ts 2.14 KiB [built]
     | ./src/builder/column/ActionsColumnBuilder.ts 1.2 KiB [built]
     | ./src/builder/column/ColumnBuilder.ts 2.23 KiB [built]
     | ./src/builder/adapter/column.ts 3.95 KiB [built]
     | ./src/builder/adapter/ranking.ts 2.01 KiB [built]
     | ./src/builder/adapter/utils.ts 622 bytes [built]
     | ./src/builder/adapter/lineup.ts 7.1 KiB [built]
     |     + 2 hidden modules
[90] ./src/ui/dialogs/index.ts 65 bytes {0} [built]
[91] ./src/ui/index.ts 527 bytes {0} [built]
[93] ./src/index.ts 1.09 KiB {0} [built]
    + 94 hidden modules

WARNING in ./src/model/NumberColumn.ts
11:0-75 "export 'default' (reexported as 'INumberColumn') was not found in './INumberColumn'
 @ ./src/model/NumberColumn.ts
 @ ./src/model/index.ts
 @ ./src/index.ts

WARNING in ./src/provider/ADataProvider.ts
12:0-41 "export 'IExportOptions' was not found in './utils'
 @ ./src/provider/ADataProvider.ts
 @ ./src/provider/index.ts
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'IAdvancedBoxPlotData' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'IBoxPlotData' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'ICategoricalBin' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'ICategoricalStatistics' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'INumberBin' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'IStatistics' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterActionsColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterCategoricalColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterDateColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterHierarchyColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterNumberColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterStringColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/dialogs/index.ts
1:0-63 "export 'IDialogOptions' was not found in './ADialog'
 @ ./src/ui/dialogs/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
1:0-70 "export 'ISidePanelOptions' was not found in './SidePanel'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IBuilderAdapter' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IBuilderAdapterDataProps' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IBuilderAdapterProps' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IChangeDetecter' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
2:0-95 "export 'IGroupSearchItem' was not found in './SearchBox'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
2:0-95 "export 'IItem' was not found in './SearchBox'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
2:0-95 "export 'ISearchBoxOptions' was not found in './SearchBox'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'ICellRenderer' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'ICellRendererFactory' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'IGroupCellRenderer' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'IImposer' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'default' (reexported as 'IRenderContext') was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/index.ts
3:0-105 "export 'IEngineRankingContext' was not found in './EngineRanking'
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/index.ts
3:0-105 "export 'IEngineRankingOptions' was not found in './EngineRanking'
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterImposeColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterNestedColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterRankingProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterReduceColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterScriptColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterSupportColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterWeightedSumColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/model/Column.ts
7:0-43 "export 'IColumnDesc' was not found in './interfaces'
 @ ./src/model/Column.ts
 @ ./src/model/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/LineUp.ts
7:0-47 "export 'ILineUpOptions' was not found in '../interfaces'
 @ ./src/ui/LineUp.ts
 @ ./src/index.ts

WARNING in ./src/ui/taggle/Taggle.ts
8:0-50 "export 'ITaggleOptions' was not found in '../../interfaces'
 @ ./src/ui/taggle/Taggle.ts
 @ ./src/ui/taggle/index.ts
 @ ./src/index.ts

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  LineUpJS.js (478 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  LineUpJS (568 KiB)
      LineUpJS.js
      LineUpJS.css


WARNING in webpack performance recommendations:
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/sass-loader/lib/loader.js!src/style.scss:
     3 assets
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [2] ./src/assets/font/lu-font.eot 82 bytes {0} [built]
    [3] ./src/assets/font/lu-font.svg 82 bytes {0} [built]
    [4] ./src/assets/font/lu-font.ttf 82 bytes {0} [built]
    [5] ./src/assets/font/lu-font.woff 15.8 KiB {0} [built]
    [6] ./src/assets/font/lu-font.woff2 13.3 KiB {0} [built]
    [9] ./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/style.scss 59.8 KiB {0} [built]
        + 4 hidden modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/sass-loader/lib/loader.js!src/styles/_vars.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [1] ./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/styles/_vars.scss 162 bytes {0} [built]        + 1 hidden module

> thinkh-lineupjs-test@3.0.1-beta.14 prepublishOnly .
> npm run build


> thinkh-lineupjs-test@3.0.1-beta.14 prebuild D:\dev\thinkh-lineupjs
> npm run clean && (node -e "process.exit(process.env.PHOVEA_SKIP_TESTS === undefined?1:0)" || npm run test)


> thinkh-lineupjs-test@3.0.1-beta.14 clean D:\dev\thinkh-lineupjs
> rimraf build dist && npm run clean:compile


> thinkh-lineupjs-test@3.0.1-beta.14 clean:compile D:\dev\thinkh-lineupjs
> rimraf src/**/*.map src/**/*.js src/**/*.d.ts tests/**/*.js tests/**/*.map tests/**/*.d.ts demo/**/*.js demo/**/*.map demo/**/*.d.ts


> thinkh-lineupjs-test@3.0.1-beta.14 pretest D:\dev\thinkh-lineupjs
> npm run clean && npm run compile


> thinkh-lineupjs-test@3.0.1-beta.14 clean D:\dev\thinkh-lineupjs
> rimraf build dist && npm run clean:compile


> thinkh-lineupjs-test@3.0.1-beta.14 clean:compile D:\dev\thinkh-lineupjs
> rimraf src/**/*.map src/**/*.js src/**/*.d.ts tests/**/*.js tests/**/*.map tests/**/*.d.ts demo/**/*.js demo/**/*.map demo/**/*.d.ts


> thinkh-lineupjs-test@3.0.1-beta.14 compile D:\dev\thinkh-lineupjs
> tsc


> thinkh-lineupjs-test@3.0.1-beta.14 test D:\dev\thinkh-lineupjs
> node -e "process.exit(require('fs').existsSync('./tests')?1:0)" || karma start


> thinkh-lineupjs-test@3.0.1-beta.14 posttest D:\dev\thinkh-lineupjs
> npm run lint


> thinkh-lineupjs-test@3.0.1-beta.14 lint D:\dev\thinkh-lineupjs
> tslint --project tsconfig.json -c tslint.json && stylelint src/**/*.scss


> thinkh-lineupjs-test@3.0.1-beta.14 build D:\dev\thinkh-lineupjs
> webpack --mode production

Starting type checking service...
Using 1 worker with 2048MB memory limit
Hash: d9d0b65fe490914dcc5b
Version: webpack 4.8.3
Time: 17469ms
Built at: 2018-06-13 17:38:48
                               Asset      Size  Chunks                    Chunk Names
248608643408d014c5148346db87791b.eot  20.5 KiB          [emitted]
b3646da949d9b249a801d1c515293f03.svg  22.6 KiB          [emitted]
b2e751c0a3ee1cfd9b19f52e9747488f.ttf  20.3 KiB          [emitted]
                         LineUpJS.js   478 KiB       0  [emitted]  [big]  LineUpJS
                        LineUpJS.css  89.5 KiB       0  [emitted]         LineUpJS
Entrypoint LineUpJS [big] = LineUpJS.js LineUpJS.css
 [4] ./src/internal/math.ts 6.81 KiB {0} [built]
[13] ./src/provider/ADataProvider.ts 19.9 KiB {0} [built]
[34] ./src/interfaces.ts 111 bytes {0} [built]
[37] ./src/ui/LineUp.ts 2.42 KiB {0} [built]
[41] ./src/provider/LocalDataProvider.ts 7.91 KiB {0} [built]
[44] ./src/renderer/index.ts + 31 modules 102 KiB {0} [built]
     | ./src/renderer/BooleanCellRenderer.ts 930 bytes [built]
     | ./src/renderer/index.ts 4.51 KiB [built]
     | ./src/renderer/StringCellRenderer.ts 5.2 KiB [built]
     | ./src/renderer/AnnotationRenderer.ts 1.39 KiB [built]
     | ./src/renderer/VerticalBarCellRenderer.ts 3.29 KiB [built]
     | ./src/renderer/BarCellRenderer.ts 2.55 KiB [built]
     | ./src/renderer/DefaultCellRenderer.ts 982 bytes [built]
     | ./src/renderer/ActionRenderer.ts 2 KiB [built]
     | ./src/renderer/BoxplotCellRenderer.ts 8.11 KiB [built]
     | ./src/renderer/BrightnessCellRenderer.ts 2.16 KiB [built]
     | ./src/renderer/CategoricalCellRenderer.ts 6.79 KiB [built]
     | ./src/renderer/UpSetCellRenderer.ts 4.68 KiB [built]
     | ./src/renderer/CategoricalHeatmapCellRenderer.ts 3.59 KiB [built]
     | ./src/renderer/CategoricalStackedDistributionlCellRenderer.ts 3.92 KiB [built]
     | ./src/renderer/CircleCellRenderer.ts 1.65 KiB [built]
     |     + 17 hidden modules
[54] ./src/model/ReduceColumn.ts 4.07 KiB {0} [built]
[68] ./src/ui/taggle/Taggle.ts 3.61 KiB {0} [built]
[75] ./src/config.ts 865 bytes {0} [built]
[79] ./src/model/index.ts + 10 modules 33.3 KiB {0} [built]
     | ./src/model/CategoricalMapColumn.ts 1.93 KiB [built]
     | ./src/model/index.ts 7.57 KiB [built]
     | ./src/model/CategoricalsColumn.ts 1.59 KiB [built]
     | ./src/model/DateColumn.ts 1.71 KiB [built]
     | ./src/model/DatesColumn.ts 3.33 KiB [built]
     | ./src/model/BooleansColumn.ts 1.4 KiB [built]
     | ./src/model/NumberMapColumn.ts 5.71 KiB [built]
     | ./src/model/SetColumn.ts 5.43 KiB [built]
     | ./src/model/StringsColumn.ts 2.25 KiB [built]
     | ./src/model/DummyColumn.ts 555 bytes [built]
     | ./src/model/DatesMapColumn.ts 1.59 KiB [built]
[86] ./src/provider/index.ts + 1 modules 3.68 KiB {0} [built]
     | ./src/provider/index.ts 337 bytes [built]
     | ./src/provider/RemoteDataProvider.ts 3.32 KiB [built]
[87] ./src/builder/index.ts + 16 modules 45 KiB {0} [built]
     | ./src/builder/column/DateColumnBuilder.ts 675 bytes [built]
     | ./src/builder/index.ts 794 bytes [built]
     | ./src/builder/DataBuilder.ts 3.84 KiB [built]
     | ./src/builder/column/index.ts 281 bytes [built]
     | ./src/builder/adapter/index.ts 1.81 KiB [built]
     | ./src/builder/column/NumberColumnBuilder.ts 3.43 KiB [built]
     | ./src/builder/column/StringColumnBuilder.ts 961 bytes [built]
     | ./src/builder/RankingBuilder.ts 10.3 KiB [built]
     | ./src/builder/LineUpBuilder.ts 2.14 KiB [built]
     | ./src/builder/column/ActionsColumnBuilder.ts 1.2 KiB [built]
     | ./src/builder/column/ColumnBuilder.ts 2.23 KiB [built]
     | ./src/builder/adapter/column.ts 3.95 KiB [built]
     | ./src/builder/adapter/ranking.ts 2.01 KiB [built]
     | ./src/builder/adapter/utils.ts 622 bytes [built]
     | ./src/builder/adapter/lineup.ts 7.1 KiB [built]
     |     + 2 hidden modules
[90] ./src/ui/dialogs/index.ts 65 bytes {0} [built]
[91] ./src/ui/index.ts 527 bytes {0} [built]
[93] ./src/index.ts 1.09 KiB {0} [built]
    + 94 hidden modules

WARNING in ./src/model/NumberColumn.ts
11:0-75 "export 'default' (reexported as 'INumberColumn') was not found in './INumberColumn'
 @ ./src/model/NumberColumn.ts
 @ ./src/model/index.ts
 @ ./src/index.ts

WARNING in ./src/provider/ADataProvider.ts
12:0-41 "export 'IExportOptions' was not found in './utils'
 @ ./src/provider/ADataProvider.ts
 @ ./src/provider/index.ts
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'IAdvancedBoxPlotData' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'IBoxPlotData' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'ICategoricalBin' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'ICategoricalStatistics' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'INumberBin' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/index.ts
13:0-135 "export 'IStatistics' was not found in './internal/math'
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterActionsColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterCategoricalColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterDateColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterHierarchyColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterNumberColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
1:0-293 "export 'IBuilderAdapterStringColumnDescProps' was not found in './column'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/dialogs/index.ts
1:0-63 "export 'IDialogOptions' was not found in './ADialog'
 @ ./src/ui/dialogs/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
1:0-70 "export 'ISidePanelOptions' was not found in './SidePanel'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IBuilderAdapter' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IBuilderAdapterDataProps' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IBuilderAdapterProps' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
2:0-108 "export 'IChangeDetecter' was not found in './lineup'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
2:0-95 "export 'IGroupSearchItem' was not found in './SearchBox'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
2:0-95 "export 'IItem' was not found in './SearchBox'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/panel/index.ts
2:0-95 "export 'ISearchBoxOptions' was not found in './SearchBox'
 @ ./src/ui/panel/index.ts
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'ICellRenderer' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'ICellRendererFactory' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'IGroupCellRenderer' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'IImposer' was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/renderer/index.ts
32:0-124 "export 'default' (reexported as 'IRenderContext') was not found in './interfaces'
 @ ./src/renderer/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/index.ts
3:0-105 "export 'IEngineRankingContext' was not found in './EngineRanking'
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/index.ts
3:0-105 "export 'IEngineRankingOptions' was not found in './EngineRanking'
 @ ./src/ui/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterImposeColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterNestedColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterRankingProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterReduceColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterScriptColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterSupportColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/builder/adapter/index.ts
3:0-266 "export 'IBuilderAdapterWeightedSumColumnProps' was not found in './ranking'
 @ ./src/builder/adapter/index.ts
 @ ./src/builder/index.ts
 @ ./src/index.ts

WARNING in ./src/model/Column.ts
7:0-43 "export 'IColumnDesc' was not found in './interfaces'
 @ ./src/model/Column.ts
 @ ./src/model/index.ts
 @ ./src/index.ts

WARNING in ./src/ui/LineUp.ts
7:0-47 "export 'ILineUpOptions' was not found in '../interfaces'
 @ ./src/ui/LineUp.ts
 @ ./src/index.ts

WARNING in ./src/ui/taggle/Taggle.ts
8:0-50 "export 'ITaggleOptions' was not found in '../../interfaces'
 @ ./src/ui/taggle/Taggle.ts
 @ ./src/ui/taggle/index.ts
 @ ./src/index.ts

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  LineUpJS.js (478 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  LineUpJS (568 KiB)
      LineUpJS.js
      LineUpJS.css


WARNING in webpack performance recommendations:
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/sass-loader/lib/loader.js!src/style.scss:
     3 assets
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [2] ./src/assets/font/lu-font.eot 82 bytes {0} [built]
    [3] ./src/assets/font/lu-font.svg 82 bytes {0} [built]
    [4] ./src/assets/font/lu-font.ttf 82 bytes {0} [built]
    [5] ./src/assets/font/lu-font.woff 15.8 KiB {0} [built]
    [6] ./src/assets/font/lu-font.woff2 13.3 KiB {0} [built]
    [9] ./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/style.scss 59.8 KiB {0} [built]
        + 4 hidden modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/sass-loader/lib/loader.js!src/styles/_vars.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [1] ./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/styles/_vars.scss 162 bytes {0} [built]
        + 1 hidden module
npm notice
npm notice package: thinkh-lineupjs-test@3.0.1-beta.14
npm notice === Tarball Contents ===
npm notice 4.8kB   package.json
npm notice 1.5kB   LICENSE
npm notice 12.6kB  README.md
npm notice 20.9kB  build/248608643408d014c5148346db87791b.eot
npm notice 20.8kB  build/b2e751c0a3ee1cfd9b19f52e9747488f.ttf
npm notice 23.1kB  build/b3646da949d9b249a801d1c515293f03.svg
npm notice 91.6kB  build/LineUpJS.css
npm notice 489.6kB build/LineUpJS.js
npm notice 17.2kB  src/assets/font/config.json
npm notice 180B    src/assets/font/LICENSE.txt
npm notice 20.9kB  src/assets/font/lu-font.eot
npm notice 23.1kB  src/assets/font/lu-font.svg
npm notice 20.8kB  src/assets/font/lu-font.ttf
npm notice 12.1kB  src/assets/font/lu-font.woff
npm notice 10.2kB  src/assets/font/lu-font.woff2
npm notice 34B     src/assets/font/README.md
npm notice 680B    src/assets/icon_categorical.svg
npm notice 902B    src/assets/icon_composite.svg
npm notice 979B    src/assets/icon_date.svg
npm notice 643B    src/assets/icon_logo.svg
npm notice 1.5kB   src/assets/icon_map.svg
npm notice 543B    src/assets/icon_matrix.svg
npm notice 987B    src/assets/icon_numerical.svg
npm notice 2.5kB   src/assets/icon_ordinal.svg
npm notice 1.4kB   src/assets/icon_slope_band.svg
npm notice 631B    src/assets/icon_slope_item.svg
npm notice 396B    src/assets/icon_string.svg
npm notice 2.5kB   src/builder/adapter/column.d.ts
npm notice 4.1kB   src/builder/adapter/column.js
npm notice 2.0kB   src/builder/adapter/index.d.ts
npm notice 1.2kB   src/builder/adapter/index.js
npm notice 2.1kB   src/builder/adapter/lineup.d.ts
npm notice 7.3kB   src/builder/adapter/lineup.js
npm notice 2.2kB   src/builder/adapter/ranking.d.ts
npm notice 2.1kB   src/builder/adapter/ranking.js
npm notice 346B    src/builder/adapter/utils.d.ts
npm notice 655B    src/builder/adapter/utils.js
npm notice 469B    src/builder/column/ActionsColumnBuilder.d.ts
npm notice 1.3kB   src/builder/column/ActionsColumnBuilder.js
npm notice 663B    src/builder/column/CategoricalColumnBuilder.d.ts
npm notice 2.8kB   src/builder/column/CategoricalColumnBuilder.js
npm notice 695B    src/builder/column/ColumnBuilder.d.ts
npm notice 2.3kB   src/builder/column/ColumnBuilder.js
npm notice 340B    src/builder/column/DateColumnBuilder.d.ts
npm notice 720B    src/builder/column/DateColumnBuilder.js
npm notice 504B    src/builder/column/HierarchyColumnBuilder.d.ts
npm notice 1.1kB   src/builder/column/HierarchyColumnBuilder.js
npm notice 281B    src/builder/column/index.d.ts
npm notice 314B    src/builder/column/index.js
npm notice 794B    src/builder/column/NumberColumnBuilder.d.ts
npm notice 3.6kB   src/builder/column/NumberColumnBuilder.js
npm notice 400B    src/builder/column/StringColumnBuilder.d.ts
npm notice 1.0kB   src/builder/column/StringColumnBuilder.js
npm notice 1.4kB   src/builder/DataBuilder.d.ts
npm notice 4.0kB   src/builder/DataBuilder.js
npm notice 395B    src/builder/index.d.ts
npm notice 827B    src/builder/index.js
npm notice 1.1kB   src/builder/LineUpBuilder.d.ts
npm notice 2.2kB   src/builder/LineUpBuilder.js
npm notice 2.1kB   src/builder/RankingBuilder.d.ts
npm notice 10.6kB  src/builder/RankingBuilder.js
npm notice 138B    src/config.d.ts
npm notice 899B    src/config.js
npm notice 1.2kB   src/index.d.ts
npm notice 1.0kB   src/index.js
npm notice 1.6kB   src/interfaces.d.ts
npm notice 149B    src/interfaces.js
npm notice 1.1kB   src/internal/AEventDispatcher.d.ts
npm notice 3.2kB   src/internal/AEventDispatcher.js
npm notice 12B     src/internal/debounce.d.ts
npm notice 769B    src/internal/debounce.js
npm notice 454B    src/internal/dnd.d.ts
npm notice 5.1kB   src/internal/dnd.js
npm notice 111B    src/internal/index.d.ts
npm notice 144B    src/internal/index.js
npm notice 795B    src/internal/math.d.ts
npm notice 7.0kB   src/internal/math.js
npm notice 12B     src/internal/merge.d.ts
npm notice 1.3kB   src/internal/merge.js
npm notice 538B    src/internal/OrderedSet.d.ts
npm notice 2.0kB   src/internal/OrderedSet.js
npm notice 12B     src/internal/utils.d.ts
npm notice 774B    src/internal/utils.js
npm notice 1.1kB   src/model/ActionColumn.d.ts
npm notice 1.2kB   src/model/ActionColumn.js
npm notice 816B    src/model/AggregateGroupColumn.d.ts
npm notice 2.2kB   src/model/AggregateGroupColumn.js
npm notice 516B    src/model/AnnotateColumn.d.ts
npm notice 2.1kB   src/model/AnnotateColumn.js
npm notice 2.5kB   src/model/annotations.d.ts
npm notice 2.4kB   src/model/annotations.js
npm notice 1.3kB   src/model/ArrayColumn.d.ts
npm notice 3.3kB   src/model/ArrayColumn.js
npm notice 1.6kB   src/model/BooleanColumn.d.ts
npm notice 4.9kB   src/model/BooleanColumn.js
npm notice 707B    src/model/BooleansColumn.d.ts
npm notice 1.5kB   src/model/BooleansColumn.js
npm notice 2.0kB   src/model/BoxPlotColumn.d.ts
npm notice 5.6kB   src/model/BoxPlotColumn.js
npm notice 1.5kB   src/model/CategoricalColumn.d.ts
npm notice 5.4kB   src/model/CategoricalColumn.js
npm notice 845B    src/model/CategoricalMapColumn.d.ts
npm notice 2.0kB   src/model/CategoricalMapColumn.js
npm notice 733B    src/model/CategoricalsColumn.d.ts
npm notice 1.7kB   src/model/CategoricalsColumn.js
npm notice 4.6kB   src/model/Column.d.ts
npm notice 14.0kB  src/model/Column.js
npm notice 1.5kB   src/model/CompositeColumn.d.ts
npm notice 6.7kB   src/model/CompositeColumn.js
npm notice 1.1kB   src/model/CompositeNumberColumn.d.ts
npm notice 2.9kB   src/model/CompositeNumberColumn.js
npm notice 580B    src/model/DateColumn.d.ts
npm notice 1.8kB   src/model/DateColumn.js
npm notice 983B    src/model/DatesColumn.d.ts
npm notice 3.5kB   src/model/DatesColumn.js
npm notice 642B    src/model/DatesMapColumn.d.ts
npm notice 1.7kB   src/model/DatesMapColumn.js
npm notice 236B    src/model/DummyColumn.d.ts
npm notice 594B    src/model/DummyColumn.js
npm notice 158B    src/model/Group.d.ts
npm notice 107B    src/model/Group.js
npm notice 521B    src/model/GroupColumn.d.ts
npm notice 2.3kB   src/model/GroupColumn.js
npm notice 2.4kB   src/model/HierarchyColumn.d.ts
npm notice 8.5kB   src/model/HierarchyColumn.js
npm notice 658B    src/model/IArrayColumn.d.ts
npm notice 315B    src/model/IArrayColumn.js
npm notice 1.7kB   src/model/ICategoricalColumn.d.ts
npm notice 3.3kB   src/model/ICategoricalColumn.js
npm notice 1.8kB   src/model/ImpositionBoxPlotColumn.d.ts
npm notice 6.9kB   src/model/ImpositionBoxPlotColumn.js
npm notice 1.2kB   src/model/ImpositionCompositeColumn.d.ts
npm notice 5.2kB   src/model/ImpositionCompositeColumn.js
npm notice 2.2kB   src/model/ImpositionCompositesColumn.d.ts
npm notice 8.4kB   src/model/ImpositionCompositesColumn.js
npm notice 7.1kB   src/model/index.d.ts
npm notice 7.8kB   src/model/index.js
npm notice 1.0kB   src/model/interfaces.d.ts
npm notice 119B    src/model/interfaces.js
npm notice 88B     src/model/internal.d.ts
npm notice 3.4kB   src/model/internal.js
npm notice 2.8kB   src/model/INumberColumn.d.ts
npm notice 2.9kB   src/model/INumberColumn.js
npm notice 645B    src/model/MapColumn.d.ts
npm notice 2.0kB   src/model/MapColumn.js
npm notice 2.2kB   src/model/MappingFunction.d.ts
npm notice 6.3kB   src/model/MappingFunction.js
npm notice 247B    src/model/missing.d.ts
npm notice 545B    src/model/missing.js
npm notice 1.1kB   src/model/MultiLevelCompositeColumn.d.ts
npm notice 4.9kB   src/model/MultiLevelCompositeColumn.js
npm notice 414B    src/model/NestedColumn.d.ts
npm notice 1.2kB   src/model/NestedColumn.js
npm notice 2.0kB   src/model/NumberColumn.d.ts
npm notice 8.9kB   src/model/NumberColumn.js
npm notice 2.0kB   src/model/NumberMapColumn.d.ts
npm notice 5.9kB   src/model/NumberMapColumn.js
npm notice 1.9kB   src/model/NumbersColumn.d.ts
npm notice 6.0kB   src/model/NumbersColumn.js
npm notice 1.7kB   src/model/OrdinalColumn.d.ts
npm notice 5.7kB   src/model/OrdinalColumn.js
npm notice 325B    src/model/RankColumn.d.ts
npm notice 946B    src/model/RankColumn.js
npm notice 3.9kB   src/model/Ranking.d.ts
npm notice 24.3kB  src/model/Ranking.js
npm notice 1.1kB   src/model/ReduceColumn.d.ts
npm notice 4.2kB   src/model/ReduceColumn.js
npm notice 987B    src/model/ScriptColumn.d.ts
npm notice 6.6kB   src/model/ScriptColumn.js
npm notice 1.0kB   src/model/SelectionColumn.d.ts
npm notice 3.0kB   src/model/SelectionColumn.js
npm notice 1.6kB   src/model/SetColumn.d.ts
npm notice 5.6kB   src/model/SetColumn.js
npm notice 1.7kB   src/model/StackColumn.d.ts
npm notice 8.8kB   src/model/StackColumn.js
npm notice 1.4kB   src/model/StringColumn.d.ts
npm notice 5.2kB   src/model/StringColumn.js
npm notice 927B    src/model/StringMapColumn.d.ts
npm notice 2.6kB   src/model/StringMapColumn.js
npm notice 842B    src/model/StringsColumn.d.ts
npm notice 2.3kB   src/model/StringsColumn.js
npm notice 843B    src/model/ValueColumn.d.ts
npm notice 2.2kB   src/model/ValueColumn.js
npm notice 1.3kB   src/provider/ACommonDataProvider.d.ts
npm notice 5.9kB   src/provider/ACommonDataProvider.js
npm notice 5.6kB   src/provider/ADataProvider.d.ts
npm notice 20.4kB  src/provider/ADataProvider.js
npm notice 337B    src/provider/index.d.ts
npm notice 370B    src/provider/index.js
npm notice 1.6kB   src/provider/LocalDataProvider.d.ts
npm notice 8.1kB   src/provider/LocalDataProvider.js
npm notice 1.4kB   src/provider/RemoteDataProvider.d.ts
npm notice 3.4kB   src/provider/RemoteDataProvider.js
npm notice 737B    src/provider/utils.d.ts
npm notice 4.3kB   src/provider/utils.js
npm notice 12B     src/renderer/AAggregatedGroupRenderer.d.ts
npm notice 929B    src/renderer/AAggregatedGroupRenderer.js
npm notice 12B     src/renderer/ActionRenderer.d.ts
npm notice 2.1kB   src/renderer/ActionRenderer.js
npm notice 12B     src/renderer/AggregateGroupRenderer.d.ts
npm notice 2.7kB   src/renderer/AggregateGroupRenderer.js
npm notice 12B     src/renderer/AnnotationRenderer.d.ts
npm notice 1.5kB   src/renderer/AnnotationRenderer.js
npm notice 145B    src/renderer/ANumbersCellRenderer.d.ts
npm notice 2.7kB   src/renderer/ANumbersCellRenderer.js
npm notice 12B     src/renderer/BarCellRenderer.d.ts
npm notice 2.6kB   src/renderer/BarCellRenderer.js
npm notice 12B     src/renderer/BooleanCellRenderer.d.ts
npm notice 977B    src/renderer/BooleanCellRenderer.js
npm notice 12B     src/renderer/BoxplotCellRenderer.d.ts
npm notice 8.4kB   src/renderer/BoxplotCellRenderer.js
npm notice 12B     src/renderer/BrightnessCellRenderer.d.ts
npm notice 2.3kB   src/renderer/BrightnessCellRenderer.js
npm notice 285B    src/renderer/CategoricalCellRenderer.d.ts
npm notice 7.0kB   src/renderer/CategoricalCellRenderer.js
npm notice 12B     src/renderer/CategoricalHeatmapCellRenderer.d.ts
npm notice 3.7kB   src/renderer/CategoricalHeatmapCellRenderer.js
npm notice 12B     src/renderer/CategoricalStackedDistributionlCellRenderer.d.ts
npm notice 4.1kB   src/renderer/CategoricalStackedDistributionlCellRenderer.js
npm notice 12B     src/renderer/CircleCellRenderer.d.ts
npm notice 1.7kB   src/renderer/CircleCellRenderer.js
npm notice 502B    src/renderer/DefaultCellRenderer.d.ts
npm notice 1.0kB   src/renderer/DefaultCellRenderer.js
npm notice 12B     src/renderer/DotCellRenderer.d.ts
npm notice 4.9kB   src/renderer/DotCellRenderer.js
npm notice 12B     src/renderer/GroupCellRenderer.d.ts
npm notice 1.5kB   src/renderer/GroupCellRenderer.js
npm notice 12B     src/renderer/HeatmapCellRenderer.d.ts
npm notice 3.2kB   src/renderer/HeatmapCellRenderer.js
npm notice 444B    src/renderer/HistogramCellRenderer.d.ts
npm notice 11.9kB  src/renderer/HistogramCellRenderer.js
npm notice 12B     src/renderer/ImageCellRenderer.d.ts
npm notice 1.2kB   src/renderer/ImageCellRenderer.js
npm notice 12B     src/renderer/impose.d.ts
npm notice 691B    src/renderer/impose.js
npm notice 1.2kB   src/renderer/index.d.ts
npm notice 4.5kB   src/renderer/index.js
npm notice 2.3kB   src/renderer/interfaces.d.ts
npm notice 294B    src/renderer/interfaces.js
npm notice 12B     src/renderer/InterleavingCellRenderer.d.ts
npm notice 4.1kB   src/renderer/InterleavingCellRenderer.js
npm notice 12B     src/renderer/LinkCellRenderer.d.ts
npm notice 2.3kB   src/renderer/LinkCellRenderer.js
npm notice 12B     src/renderer/LinkMapCellRenderer.d.ts
npm notice 3.1kB   src/renderer/LinkMapCellRenderer.js
npm notice 12B     src/renderer/LoadingCellRenderer.d.ts
npm notice 775B    src/renderer/LoadingCellRenderer.js
npm notice 12B     src/renderer/MapBarCellRenderer.d.ts
npm notice 2.4kB   src/renderer/MapBarCellRenderer.js
npm notice 462B    src/renderer/missing.d.ts
npm notice 913B    src/renderer/missing.js
npm notice 12B     src/renderer/MultiLevelCellRenderer.d.ts
npm notice 7.0kB   src/renderer/MultiLevelCellRenderer.js
npm notice 12B     src/renderer/RankCellRenderer.d.ts
npm notice 1.6kB   src/renderer/RankCellRenderer.js
npm notice 12B     src/renderer/SelectionRenderer.d.ts
npm notice 4.1kB   src/renderer/SelectionRenderer.js
npm notice 12B     src/renderer/SparklineCellRenderer.d.ts
npm notice 2.8kB   src/renderer/SparklineCellRenderer.js
npm notice 12B     src/renderer/StringCellRenderer.d.ts
npm notice 5.4kB   src/renderer/StringCellRenderer.js
npm notice 127B    src/renderer/TableCellRenderer.d.ts
npm notice 4.7kB   src/renderer/TableCellRenderer.js
npm notice 175B    src/renderer/UpSetCellRenderer.d.ts
npm notice 4.8kB   src/renderer/UpSetCellRenderer.js
npm notice 184B    src/renderer/utils.d.ts
npm notice 3.7kB   src/renderer/utils.js
npm notice 12B     src/renderer/VerticalBarCellRenderer.d.ts
npm notice 3.4kB   src/renderer/VerticalBarCellRenderer.js
npm notice 25B     src/style.scss
npm notice 1.1kB   src/styles/_common.scss
npm notice 1.6kB   src/styles/_datatypes.scss
npm notice 4.3kB   src/styles/_dialogs.scss
npm notice 571B    src/styles/_lineup.scss
npm notice 2.1kB   src/styles/_mappingeditor.scss
npm notice 1.6kB   src/styles/_slope_graph.scss
npm notice 2.4kB   src/styles/_vars.scss
npm notice 2.8kB   src/styles/engine/_header.scss
npm notice 292B    src/styles/engine/_index.scss
npm notice 1.0kB   src/styles/engine/_row.scss
npm notice 503B    src/styles/engine/_selection.scss
npm notice 1.0kB   src/styles/header/_index.scss
npm notice 540B    src/styles/header/_summary.scss
npm notice 4.2kB   src/styles/header/_toolbar.scss
npm notice 572B    src/styles/icons/_custom_font.scss
npm notice 2.0kB   src/styles/icons/_font_awesome.scss
npm notice 406B    src/styles/icons/_icon_style.scss
npm notice 2.5kB   src/styles/icons/_index.scss
npm notice 726B    src/styles/index.d.ts
npm notice 2.0kB   src/styles/index.js
npm notice 1.1kB   src/styles/panel/_entry.scss
npm notice 1.6kB   src/styles/panel/_index.scss
npm notice 1.7kB   src/styles/panel/_search.scss
npm notice 1.7kB   src/styles/renderer/_aggregate.scss
npm notice 625B    src/styles/renderer/_bar.scss
npm notice 2.1kB   src/styles/renderer/_boxplot.scss
npm notice 362B    src/styles/renderer/_catdistributionbar.scss
npm notice 606B    src/styles/renderer/_categorical.scss
npm notice 402B    src/styles/renderer/_dot.scss
npm notice 627B    src/styles/renderer/_heatmap.scss
npm notice 3.0kB   src/styles/renderer/_histogram.scss
npm notice 532B    src/styles/renderer/_image.scss
npm notice 517B    src/styles/renderer/_index.scss
npm notice 132B    src/styles/renderer/_interleaving.scss
npm notice 350B    src/styles/renderer/_missing.scss
npm notice 280B    src/styles/renderer/_rank.scss
npm notice 593B    src/styles/renderer/_selection.scss
npm notice 239B    src/styles/renderer/_sparkline.scss
npm notice 80B     src/styles/renderer/_stack.scss
npm notice 220B    src/styles/renderer/_string.scss
npm notice 429B    src/styles/renderer/_table.scss
npm notice 138B    src/styles/renderer/_threshold.scss
npm notice 602B    src/styles/renderer/_upset.scss
npm notice 97B     src/styles/renderer/_verticalbar.scss
npm notice 106B    src/styles/taggle/_index.scss
npm notice 2.6kB   src/styles/taggle/_lod.scss
npm notice 1.5kB   src/styles/taggle/_switcher.scss
npm notice 1.2kB   src/ui/ALineUp.d.ts
npm notice 2.7kB   src/ui/ALineUp.js
npm notice 12B     src/ui/animation.d.ts
npm notice 3.6kB   src/ui/animation.js
npm notice 1.2kB   src/ui/dialogs/ADialog.d.ts
npm notice 4.1kB   src/ui/dialogs/ADialog.js
npm notice 12B     src/ui/dialogs/BooleanFilterDialog.d.ts
npm notice 2.2kB   src/ui/dialogs/BooleanFilterDialog.js
npm notice 12B     src/ui/dialogs/CategoricalFilterDialog.d.ts
npm notice 3.0kB   src/ui/dialogs/CategoricalFilterDialog.js
npm notice 12B     src/ui/dialogs/CategoricalMappingFilterDialog.d.ts
npm notice 4.2kB   src/ui/dialogs/CategoricalMappingFilterDialog.js
npm notice 12B     src/ui/dialogs/ChangeRendererDialog.d.ts
npm notice 2.9kB   src/ui/dialogs/ChangeRendererDialog.js
npm notice 12B     src/ui/dialogs/CompositeChildrenDialog.d.ts
npm notice 2.4kB   src/ui/dialogs/CompositeChildrenDialog.js
npm notice 12B     src/ui/dialogs/CutOffHierarchyDialog.d.ts
npm notice 2.8kB   src/ui/dialogs/CutOffHierarchyDialog.js
npm notice 719B    src/ui/dialogs/DialogManager.d.ts
npm notice 4.2kB   src/ui/dialogs/DialogManager.js
npm notice 12B     src/ui/dialogs/EditPatternDialog.d.ts
npm notice 1.7kB   src/ui/dialogs/EditPatternDialog.js
npm notice 65B     src/ui/dialogs/index.d.ts
npm notice 82B     src/ui/dialogs/index.js
npm notice 12B     src/ui/dialogs/InputNumberDialog.d.ts
npm notice 1.4kB   src/ui/dialogs/InputNumberDialog.js
npm notice 12B     src/ui/dialogs/MappingLineDialog.d.ts
npm notice 6.0kB   src/ui/dialogs/MappingLineDialog.js
npm notice 12B     src/ui/dialogs/MappingsFilterDialog.d.ts
npm notice 13.8kB  src/ui/dialogs/MappingsFilterDialog.js
npm notice 12B     src/ui/dialogs/MoreColumnOptionsDialog.d.ts
npm notice 847B    src/ui/dialogs/MoreColumnOptionsDialog.js
npm notice 12B     src/ui/dialogs/ReduceDialog.d.ts
npm notice 916B    src/ui/dialogs/ReduceDialog.js
npm notice 12B     src/ui/dialogs/RenameDialog.d.ts
npm notice 1.8kB   src/ui/dialogs/RenameDialog.js
npm notice 12B     src/ui/dialogs/ScriptEditDialog.d.ts
npm notice 1.1kB   src/ui/dialogs/ScriptEditDialog.js
npm notice 12B     src/ui/dialogs/SearchDialog.d.ts
npm notice 1.7kB   src/ui/dialogs/SearchDialog.js
npm notice 12B     src/ui/dialogs/SortDateDialog.d.ts
npm notice 711B    src/ui/dialogs/SortDateDialog.js
npm notice 12B     src/ui/dialogs/SortDialog.d.ts
npm notice 2.8kB   src/ui/dialogs/SortDialog.js
npm notice 12B     src/ui/dialogs/SortGroupDialog.d.ts
npm notice 687B    src/ui/dialogs/SortGroupDialog.js
npm notice 12B     src/ui/dialogs/StratifyThresholdDialog.d.ts
npm notice 1.9kB   src/ui/dialogs/StratifyThresholdDialog.js
npm notice 12B     src/ui/dialogs/StringFilterDialog.d.ts
npm notice 3.3kB   src/ui/dialogs/StringFilterDialog.js
npm notice 73B     src/ui/dialogs/utils.d.ts
npm notice 596B    src/ui/dialogs/utils.js
npm notice 12B     src/ui/dialogs/WeightsEditDialog.d.ts
npm notice 2.1kB   src/ui/dialogs/WeightsEditDialog.js
npm notice 3.9kB   src/ui/EngineRanking.d.ts
npm notice 31.1kB  src/ui/EngineRanking.js
npm notice 2.0kB   src/ui/EngineRenderer.d.ts
npm notice 17.9kB  src/ui/EngineRenderer.js
npm notice 559B    src/ui/header.d.ts
npm notice 18.9kB  src/ui/header.js
npm notice 527B    src/ui/index.d.ts
npm notice 429B    src/ui/index.js
npm notice 1.5kB   src/ui/interfaces.d.ts
npm notice 38B     src/ui/interfaces.js
npm notice 694B    src/ui/LineUp.d.ts
npm notice 2.5kB   src/ui/LineUp.js
npm notice 81B     src/ui/missing.d.ts
npm notice 1.4kB   src/ui/missing.js
npm notice 12B     src/ui/MultiLevelRenderColumn.d.ts
npm notice 4.5kB   src/ui/MultiLevelRenderColumn.js
npm notice 169B    src/ui/panel/index.d.ts
npm notice 139B    src/ui/panel/index.js
npm notice 1.1kB   src/ui/panel/SearchBox.d.ts
npm notice 7.0kB   src/ui/panel/SearchBox.js
npm notice 1.1kB   src/ui/panel/SidePanel.d.ts
npm notice 12.6kB  src/ui/panel/SidePanel.js
npm notice 747B    src/ui/panel/SidePanelEntry.d.ts
npm notice 1.5kB   src/ui/panel/SidePanelEntry.js
npm notice 12B     src/ui/panel/SidePanelEntryVis.d.ts
npm notice 2.0kB   src/ui/panel/SidePanelEntryVis.js
npm notice 1.4kB   src/ui/RenderColumn.d.ts
npm notice 5.2kB   src/ui/RenderColumn.js
npm notice 12B     src/ui/SelectionManager.d.ts
npm notice 6.0kB   src/ui/SelectionManager.js
npm notice 1.7kB   src/ui/SlopeGraph.d.ts
npm notice 16.6kB  src/ui/SlopeGraph.js
npm notice 237B    src/ui/taggle/index.d.ts
npm notice 200B    src/ui/taggle/index.js
npm notice 446B    src/ui/taggle/interfaces.d.ts
npm notice 38B     src/ui/taggle/interfaces.js
npm notice 180B    src/ui/taggle/spaceFillingRule.d.ts
npm notice 2.3kB   src/ui/taggle/spaceFillingRule.js
npm notice 779B    src/ui/taggle/Taggle.d.ts
npm notice 3.7kB   src/ui/taggle/Taggle.js
npm notice 1.6kB   src/ui/taggle/TaggleRenderer.d.ts
npm notice 5.6kB   src/ui/taggle/TaggleRenderer.js
npm notice 866B    src/ui/toolbar.d.ts
npm notice 8.0kB   src/ui/toolbar.js
npm notice === Tarball Details ===
npm notice name:          thinkh-lineupjs-test
npm notice version:       3.0.1-beta.14
npm notice package size:  408.2 kB
npm notice unpacked size: 1.6 MB
npm notice shasum:        0246643b1cf230c8105ab775dc3444e819df95c6
npm notice integrity:     sha512-y/IkAD84cSSRg[...]UH6ei6Fq2m49w==
npm notice total files:   415
npm notice
+ thinkh-lineupjs-test@3.0.1-beta.14
Counting objects: 5, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 540 bytes | 540.00 KiB/s, done.
Total 5 (delta 3), reused 0 (delta 0)
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To https://github.com/thinkh/lineupjs.git
   9827b6b1..26552c9d  thinkh/npm-dependency-test -> thinkh/npm-dependency-test
 * [new tag]           v3.0.1-beta.14 -> v3.0.1-beta.14
PS D:\dev\thinkh-lineupjs>
```

sgratzl commented:
seems like the `prepublishOnly` step is then superfluous.

thinkh commented:
Yes, correct. Then I'll replace `prepublishOnly` with `prepare`. Ok?

sgratzl commented:
based on your tests I don't see any harm that speaks against it

thinkh commented:
Great news! 👍
